pilot/pkg/model/service.go

```diff
@@ -481,8 +483,24 @@ type IstioEndpoint struct {
 	// Labels points to the workload or deployment labels.
 	Labels labels.Instance
 
-	// Address is the address of the endpoint, using envoy proto.
-	Address string
+	// Addresses is the address of the endpoint, using envoy proto:
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint_components.proto#config-endpoint-v3-endpoint-additionaladdress
+	// This change can support multiple addresses for an endpoint, then there are 2 use cases for it, such as
+	// 1. An endpoint can have both ipv4 or ipv6.
+	// 2. An endpoint can have multiple addresses, more then one IP addresses which are the same IP families.
```

Comment:
How can this happen in Istio?

Reply:
Hi @howardjohn, besides the Dual Stack scenario, it can be helpful for headless service which may represent a set of endpoints. So one headless service can have a endpoint with multiple addresses. Of course, the happy eye bulls way can also handle this situation.